dandi/dandiapi.py

```diff
@@ -260,6 +260,12 @@ def __init__(self, api_url, token=None):
 
     def authenticate(self, token):
         self._headers["Authorization"] = f"token {token}"
+        try:
+            # Fails if token is invalid:
+            self.get("/auth/token")
+        except requests.HTTPError:
+            del self._headers["Authorization"]
+            raise
```

Comment:
would it possible to instrument some test for this to ensure that we do prompt user to enter a new token?

Reply:
We don't prompt the user for a new token.  Failure has the same result as it did for Girder: an exception is raised, and the program exits.

Reply:
well -- we should at least here in `dandi_authenticate`: https://github.com/dandi/dandi-cli/blob/HEAD/dandi/dandiapi.py#L276 where we first ask for the key if not known and then proceed to self.authenticate.  We should make it into a loop which would try to authenticate, and if fails - tell user that the key is not good, would they like to enter a new one -- if say 'y' -- go back into the loop prompting for `Please provide API Key ...` and if 'n' -- reraise

Reply:
clarification: loop/ask only if `dandi/utils.py:def is_interactive():`

Reply:
Done.